07_distributed_systems/01_intro/client/client.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"kv_store/protocol"
+	"net/http"
+	"os"
+	"regexp"
+	"time"
+)
+
+const (
+	SERVER = "127.0.0.1"
+	PORT   = "8000"
+	URL    = "http://" + SERVER + ":" + PORT
+)
+
+func panicOnError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// abozhenko for oz: Is it ok to use http at all?
+
+func get(key string) (response string, err error) {
+	//TODO: re-use tcp connection for entire repl session?
+	getRequest := protocol.GetRequest{
+		Key: []byte(key),
+	}
+	reqBytes := getRequest.Encode()
+
+	//abozhenko for oz: Do you think it is ok to use get request body?
```

Comment:
Yes I think it's fine if you are defining the protocol. There's a chance that other systems (e.g. proxies/caches etc) would ignore the body, but that's unlikely to come up here).